From 3ba56ede807c3995b4b94e88c8ba3aa5c9db913b Mon Sep 17 00:00:00 2001
From: cnauroth <cnauroth@apache.org>
Date: Fri, 3 Apr 2015 10:44:02 -0700
Subject: [PATCH 1946/2748] HDFS-8039. Fix TestDebugAdmin#testRecoverLease and
 testVerfiyBlockChecksumCommand on Windows.
 Contributed by Xiaoyu Yao.

(cherry picked from commit db80e428919e4e763f56b9344c748e7f8fb46491)
(cherry picked from commit 28e0602e7bf20581e3ce477512c60a854a4612d3)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: I87c8a754f42e9dad284f4717f37090b23e659f4e
---
 .../apache/hadoop/hdfs/tools/TestDebugAdmin.java   |   15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDebugAdmin.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDebugAdmin.java
index 44b6ba9..52b194d 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDebugAdmin.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDebugAdmin.java
@@ -77,17 +77,18 @@ private String runCmd(String[] cmd) throws Exception {
       System.setOut(oldOut);
       IOUtils.closeStream(out);
     }
-    return "ret: " + ret + ", " + bytes.toString();
+    return "ret: " + ret + ", " +
+        bytes.toString().replaceAll(System.getProperty("line.separator"), "");
   }
 
   @Test(timeout = 60000)
   public void testRecoverLease() throws Exception {
-    assertEquals("ret: 1, You must supply a -path argument to recoverLease.\n",
+    assertEquals("ret: 1, You must supply a -path argument to recoverLease.",
         runCmd(new String[]{"recoverLease", "-retries", "1"}));
     FSDataOutputStream out = fs.create(new Path("/foo"));
     out.write(123);
     out.close();
-    assertEquals("ret: 0, recoverLease SUCCEEDED on /foo\n",
+    assertEquals("ret: 0, recoverLease SUCCEEDED on /foo",
         runCmd(new String[]{"recoverLease", "-path", "/foo"}));
   }
 
@@ -98,18 +99,18 @@ public void testVerifyBlockChecksumCommand() throws Exception {
     ExtendedBlock block = DFSTestUtil.getFirstBlock(fs, new Path("/bar"));
     File blockFile = getBlockFile(fsd,
         block.getBlockPoolId(), block.getLocalBlock());
-    assertEquals("ret: 1, You must specify a meta file with -meta\n",
+    assertEquals("ret: 1, You must specify a meta file with -meta",
         runCmd(new String[]{"verify", "-block", blockFile.getAbsolutePath()}));
     File metaFile = getMetaFile(fsd,
         block.getBlockPoolId(), block.getLocalBlock());
     assertEquals("ret: 0, Checksum type: " +
-          "DataChecksum(type=CRC32C, chunkSize=512)\n",
+          "DataChecksum(type=CRC32C, chunkSize=512)",
         runCmd(new String[]{"verify",
             "-meta", metaFile.getAbsolutePath()}));
     assertEquals("ret: 0, Checksum type: " +
-          "DataChecksum(type=CRC32C, chunkSize=512)\n" +
+          "DataChecksum(type=CRC32C, chunkSize=512)" +
           "Checksum verification succeeded on block file " +
-          blockFile.getAbsolutePath() + "\n",
+          blockFile.getAbsolutePath(),
         runCmd(new String[]{"verify",
             "-meta", metaFile.getAbsolutePath(),
             "-block", blockFile.getAbsolutePath()})
-- 
1.7.9.5

